monai/transforms/io/dictionary.py

```diff
@@ -211,41 +216,47 @@ class SaveImaged(MapTransform):
             folders with the same file names. For example, with the following inputs:
 
-            - input_file_name: `/foo/bar/test1/image.nii`
-            - output_postfix: `seg`
-            - output_ext: `.nii.gz`
-            - output_dir: `/output`
-            - data_root_dir: `/foo/bar`
+            - input_file_name: ``/foo/bar/test1/image.nii``
+            - output_postfix: ``seg``
+            - output_ext: ``.nii.gz``
+            - output_dir: ``/output``
+            - data_root_dir: ``/foo/bar``
 
-            The output will be: /output/test1/image/image_seg.nii.gz
+            The output will be: ``/output/test1/image/image_seg.nii.gz``
 
+            Handled by ``folder_layout`` instead, if ``folder_layout`` not ``None``.
         separate_folder: whether to save every file in a separate folder. For example: for the input filename
-            `image.nii`, postfix `seg` and folder_path `output`, if `separate_folder=True`, it will be saved as:
-            `output/image/image_seg.nii`, if `False`, saving as `output/image_seg.nii`. Default to `True`.
-        print_log: whether to print logs when saving. Default to `True`.
+            ``image.nii``, postfix ``seg`` and folder_path ``output``, if ``separate_folder=True``, it will be saved as:
+            ``output/image/image_seg.nii``, if ``False``, saving as ``output/image_seg.nii``. Default to ``True``.
+            Handled by ``folder_layout`` instead, if ``folder_layout`` not ``None``.
+        print_log: whether to print logs when saving. Default to ``True``.
         output_format: an optional string to specify the output image writer.
-            see also: `monai.data.image_writer.SUPPORTED_WRITERS`.
-        writer: a customised `monai.data.ImageWriter` subclass to save data arrays.
-            if `None`, use the default writer from `monai.data.image_writer` according to `output_ext`.
+            see also: ``monai.data.image_writer.SUPPORTED_WRITERS``.
+        writer: a customised ``monai.data.ImageWriter`` subclass to save data arrays.
+            if ``None``, use the default writer from ``monai.data.image_writer`` according to ``output_ext``.
             if it's a string, it's treated as a class name or dotted path;
             the supported built-in writer classes are ``"NibabelWriter"``, ``"ITKWriter"``, ``"PILWriter"``.
         output_name_formatter: a callable function (returning a kwargs dict) to format the output file name.
             see also: :py:func:`monai.data.folder_layout.default_name_formatter`.
-
+            If using a custom ``folder_layout``, consider providing your own formatter.
+        folder_layout: A customized ``monai.data.FolderLayoutBase`` subclass to define file naming schemes.
+            if ``None``, uses the default ``FolderLayout``
+        savepath_in_metadict: if ``True``, adds a key ``saved_to`` to the metadata, which contains the path
```

Comment:
May I know the use case when we need a `save_to` key in metadata? Does `print_log` not enough?
I am concerned when users pre-processing the same data twice and saved it twice, the `save_to` may be confused here?

Reply:
`print_log` only, well, prints. If I want to do something with the path to where the images were saved I'd need to either parse the output logs on the fly, or rerun the `FolderLayout` code to get the path. It would be a lot more efficient and less prone to error to save the output path to a variable.

My use case was that I'm trying to set up an experiment across several datasets, which has me create `Dataset` objects. Each of these objects holds a list of paths for the raw data. After preprocessing these paths, I want to add the outputs of the `SaveImaged` to another list in the dataset which contains the locations of the saved images. Currently, there is no simple way to either get the path, or infer the path. Adding `"save_to"` to the dict would at least simplify things.

For more details, see #6188

Reply:
Hi @MathijsdeBoer, I may get your points.
In your case, you want to prepare a dataset from several sides, but I think the output path is been standardized (such as "E:\...\Datasets\BraTS2021\Preprocessed")? And which is specified in the "output_dir"? I would like to say I didn't find the benefits from this "save_to" but may get some confusion when you want to preprocess the same data twice. 

But it's okay if you all think this arg is ok. cc @wyli @Nic-Ma

Reply:
How else would you suggest I programmatically store the output of `SaveImage` or `SaveImaged` in a list if none of the outputs actually provide this information?

As for preprocessing twice, It could be changed to be `"saved_to_{n}"`, where `n` is the nth time this particular `MetaTensor` has been saved?

Reply:
I think the current PR's implementation is ok for now, another idea is to set the path string as a property of SaveImage, then after saving we can get the path by `saver.last_saved_filename` cc @ericspod